### DISCUSSION TITLE
rspamd supports odbc

### DISCUSSION THREAD

**Discussion by eosps** [rspamd/rspamd#5122] [Ideas]
I want to write a Lua script to let rspamd write the information of the email, such as title, body, date, etc., into the database through ODBC when processing the email. In this way, the email can be quickly retrieved based on keywords. However, I found that rspamd does not seem to support regular databases. How can this be achieved?

**Comment by fatalbanana** [comment]
Perhaps what you might rather want is [full text search in your IMAP server](https://doc.dovecot.org/2.3/configuration_manual/fts/).

For storing metadata the recommended approach is to use [Clickhouse](https://rspamd.com/doc/modules/clickhouse.html).

If you need ODBC you might push information to an external service using [metadata exporter](https://rspamd.com/doc/modules/metadata_exporter.html) / your own rules; or process Rspamd's logs / do it via the MTA.

**Comment by eosps** [comment]
Thanks for your reply. What I want to do is to archive the mail. I will use the speed advantage of rspamd to structure the mail data and then write it into the database. At the same time, the mail files are stored on the disk. If I want to reply to a mail, I can quickly find the mail by searching the database and do subsequent processing.
